java/lance-jni/src/blocking_dataset.rs

```diff
@@ -0,0 +1,36 @@
+use arrow::ffi_stream::ArrowArrayStreamReader;
+use tokio::runtime::Runtime;
+
+use lance::Dataset;
+use lance::Result;
+
+pub struct BlockingDataset {
+    inner: Dataset,
+    rt: Runtime,
+}
+
+impl BlockingDataset {
+    pub fn write(reader: ArrowArrayStreamReader, uri: &str) -> Result<Self> {
```

Comment:
Is async calls not encouraged cross JNI?

Reply:
I'm just thinking async call in jni might introduce complexities.  Also I prefer to rely on the thread management in java compute engine (I'm still targeting to implement a lance connector for Trino/Presto).  But anyway, async api will be more efficient in many places,  I think we could implement the async jni later, can we?

Reply:
hey @beinan Any plans to add a Trino connector for this? If so, any sense of ETA? Thanks for building this :)